include/mbedtls/platform_util.h

```diff
@@ -55,6 +66,38 @@ extern "C" {
  */
 void mbedtls_platform_zeroize( void *buf, size_t len );
 
+#if defined(MBEDTLS_HAVE_TIME_DATE)
+/**
+ * \brief      Thread safe implementation of gmtime()
+ *
+ *             The function is an abstraction that when called behaves similar
+ *             to the gmtime() function from the C standard, but is thread
+ *             safe.
+ *
+ *             Mbed TLS will try to identify the underlying platform and
+ *             configure an appropriate underlying implementation (e.g.
+ *             gmtime_r() for POSIX and gmtime_s() for Windows). If this is
+ *             not possible, then gmtime() will be used. In this case, calls
+ *             from the library to gmtime() will be guarded by the mutex
+ *             mbedtls_threading_gmtime_mutex if MBEDTLS_THREADING_C is
+ *             enabled. It is recommended that calls from outside the library
+ *             are also guarded by this mutex.
+ *
+ *             If MBEDTLS_PLATFORM_GMTIME_R_ALT is defined, then Mbed TLS will
+ *             unconditionally use the alternative implementation for
+ *             mbedtls_platform_gmtime_r() supplied by the user at compile time.
+ *
+ * \param tt     Pointer to an object containing time (in seconds) since the
+ *               epoch to be converted
+ * \param tm_buf Pointer to an object where the results will be stored
+ *
+ * \return      Pointer to an object of type struct tm on success, otherwise
+ *              NULL
+ */
+struct tm *mbedtls_platform_gmtime_r( const mbedtls_time_t *tt,
```

Comment:
This is the Unix/POSIX `gmtime_r`, which is widely available in practice but not standard C. Since we're introducing a new API, maybe we should instead pick `gmtime_s`, which is standard C since C11, even though it isn't widely available? It doesn't matter much since the conversion between `gmtime_r` and `gmtime_s` is very simple.

Reply:
I stuck to `gmtime_r` for now to not deviate too far from the state of the previous approvals - I have no strong opinion on using `gmtime_s` vs. `gmtime_r`.